src/pybind/mgr/dashboard/services/cluster.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+from enum import Enum
+
+from .. import mgr
+
+
+class ClusterStatus(Enum):
+    INSTALLED = 0
+    POST_INSTALLED = 1
+
+
+class ClusterService(object):
+
+    @staticmethod
+    def fetch_status():
+        return mgr.get_store('cluster/status', default=ClusterStatus.INSTALLED.name)
```

Comment:
Instead of having a `/cluster/status` let's assume we just have the `Cluster` Resource (located at `/cluster`) and the status is a property of that Cluster Resource.

As you basically need a class to read (GET) and store (PUT), you just need to implement an interface that:
* Returns a dict-like representation of the cluster with a single property called `status`.
* Stores a dict-like representation of the cluster with the same property.

```py
class Cluster:
  class Status(Enum): # We don't call it ClusterStatus because it's under the Cluster namespace
    INSTALLED = 0
    POSTINSTALLED = 1

  status: Status = Status.INSTALLED

  def __init__(self, status=Status.INSTALLED): # default value
    self.status = status

  def dict(self):
    return {status: self.status}

  def to_db(self):
    mgr.set_store('cluster/status', ...)

  @classmethod
  def from_db(cls):
    return cls(status=mgr.get_store('cluster/status', ...))
```

The `dict()` method follows the [Pydantic way of doing this](https://pydantic-docs.helpmanual.io/usage/models/#model-properties) (`dataclasses` provide an [`asdict()` helper instead](https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict)).

We could make that function more smart (auto-generate `__init__` and `dict` but that's what the `dataclasses` or `pydantic.BaseModel` would do for us.

To use those we just need:
```py
# controller/cluster.py
def list(self):
  return Model.from_db().dict()

...
def set(self, cluster: Dict):
  Cluster(**cluster).to_db()
```

Reply:
Done. In PUT method (or set()) I've used str instead of Dict.